sbor/src/codec/array.rs

```diff
@@ -0,0 +1,72 @@
+use crate::rust::mem::MaybeUninit;
+use crate::rust::ptr::copy;
+use crate::rust::vec::Vec;
+use crate::type_id::*;
+use crate::*;
+
+impl<X: CustomTypeId, T: Encode<X> + TypeId<X>> Encode<X> for [T] {
+    #[inline]
+    fn encode_type_id(&self, encoder: &mut Encoder<X>) {
+        encoder.write_type_id(Self::type_id());
+    }
+    #[inline]
+    fn encode_body(&self, encoder: &mut Encoder<X>) {
+        encoder.write_type_id(T::type_id());
+        encoder.write_size(self.len());
+        if T::type_id() == SborTypeId::U8 || T::type_id() == SborTypeId::I8 {
+            let mut buf = Vec::<u8>::with_capacity(self.len());
```

Comment:
Maybe this could be done without additional allocation and copy using something like this:
let ptr = self.as_ptr().cast::<u8>();
unsafe {
    std::slice::from_raw_parts(ptr, self.len())
}

Reply:
I think I ran into some memory issue before with this approach, but let me try it again. Thanks!

Reply:
Applied. Seems to be much better :)